lib/spack/spack/cmd/ci.py

```diff
@@ -414,6 +442,9 @@ def ci_rebuild(args):
 
     install_args.extend([
         'install',
+        '--only', 'package',    # ONLY install the job spec
```

Comment:
Telling spack *not* to install dependencies in this context seems odd.  This is the reason every job in your pipeline (every job that actually does a rebuild) fails with `Error: Detected uninstalled dependencies for <spec>`.  These jobs run in a clean container with no install tree hanging around from jobs in previous stages, so dependencies *must* be installed (from buildcache populated in previous stages).

Reply:
It looked to me like the dependencies were being installed in previous stages even with this option set on the `rebuild` and this was intended to prevent them from being re-built.

I just pushed a change to remove this option.